src/ssh/allowed_signer.rs

```diff
@@ -0,0 +1,218 @@
+use std::collections::VecDeque;
+use std::fmt;
+use std::fs::File;
+use std::io::Read;
+use std::path::Path;
+
+use super::pubkey::PublicKey;
+use crate::{error::Error, Result};
+
+/// A type which represents an allowed signer entry.
+/// Please refer to [ssh-keygen-1.ALLOWED_SIGNERS] for more details about the format.
+/// [ssh-keygen-1.ALLOWED_SIGNERS]: https://man.openbsd.org/ssh-keygen.1#ALLOWED_SIGNERS
+#[derive(Debug, PartialEq, Eq)]
+pub struct AllowedSigner {
+    /// A list of principals, each in the format USER@DOMAIN.
+    pub principals: Vec<String>,
+
+    /// Indicates that this key is accepted as a CA.
+    /// This is a standard option.
+    pub cert_authority: bool,
+
+    /// Specifies a list of namespaces that are accepted for this key.
+    /// This is a standard option.
+    ///
+    /// Note: The specification allows spaces inside double quotes. However, this is not supported.
+    pub namespaces: Option<Vec<String>>,
+
+    /// Time at or after which the key is valid.
+    /// This is a standard option.
+    pub valid_after: Option<u64>,
+
+    /// Time at or before which the key is valid.
+    /// This is a standard option.
+    pub valid_before: Option<u64>,
+
+    /// Public key of the entry.
+    pub key: PublicKey,
+}
+
+/// A type which represents a collection of allowed signer entries.
+/// Please refer to [ssh-keygen-1.ALLOWED_SIGNERS] for more details about the format.
+/// [ssh-keygen-1.ALLOWED_SIGNERS]: https://man.openbsd.org/ssh-keygen.1#ALLOWED_SIGNERS
+#[derive(Debug, PartialEq, Eq)]
+pub struct AllowedSigners {
```

Comment:
You might just want to use a typedef here (I don't think there would ever be additional fields in this struct.

Reply:
I can't implement foreign traits on typedef since typedef is a weak symlink. So I chose to wrap `AllowedSigners(Vec<AllowedSigner>)`